### PR TITLE
[MRG] ENH: Keep a log of auto-detected bad channels

### PR DESCRIPTION
**Pull request by hoechenberger**
- It's good to have
- I need it for empty-room handling

To give you an example: with the `ds000248` dataset, it produces a file:
`sub-01/meg/sub-01_task-audiovisual_run-01_bad_chans.tsv` with the following content:

| Channel  | Reason                                           |
|----------|--------------------------------------------------|
| EEG 053  | pre-existing (before mne-study-template was run) |
| MEG 1032 | auto-noisy                                       |
| MEG 2313 | auto-noisy                                       |
| MEG 2443 | pre-existing (before mne-study-template was run) |

### CODE REVIEW

File: 01-import_and_maxfilter.py [hunk 1]
```diff
@@ -137,8 +137,45 @@ def find_bad_channels(raw, subject, session):
...
+        reasons.extend(['auto-noisy'] * len(auto_noisy_chs))
+        preexisting_bads = set(preexisting_bads) - set(auto_noisy_chs)
+
+    preexisting_bads = list(preexisting_bads)
+    if preexisting_bads:
+        bads_for_tsv.extend(preexisting_bads)
+        reasons.extend(['pre-existing (before mne-study-template was run)'] *
+                       len(preexisting_bads))
+
+    tsv_data = pd.DataFrame(dict(Channel=bads_for_tsv, Reason=reasons))
```
Comment (agramfort) [inline 01-import_and_maxfilter.py:174]:
```suggestion
    tsv_data = pd.DataFrame(dict(name=bads_for_tsv, reason=reasons))
```
to follow the convention of _channels.tsv files.

File: 01-import_and_maxfilter.py [hunk 2]
```diff
@@ -137,8 +137,45 @@ def find_bad_channels(raw, subject, session):
...
+    tsv_data = tsv_data.sort_values(by='Channel')
```
Comment (agramfort) [inline 01-import_and_maxfilter.py:175]:
```suggestion
    tsv_data = tsv_data.sort_values(by='name')
```

File: 01-import_and_maxfilter.py [hunk 3]
```diff
@@ -137,8 +137,45 @@ def find_bad_channels(raw, subject, session):
...
+    bids_basename = make_bids_basename(subject=subject,
+                                       session=session,
+                                       task=config.get_task(),
+                                       acquisition=config.acq,
+                                       run=run,
+                                       processing=config.proc,
+                                       recording=config.rec,
+                                       space=config.space)
+
+    bads_tsv_fname = op.join(deriv_path, f'{bids_basename}_bad_chans.tsv')
```
Comment (agramfort) [inline 01-import_and_maxfilter.py:154]:
```suggestion
    bads_tsv_fname = op.join(deriv_path, f'{bids_basename}_bad_channels.tsv')
```

**Comment by SophieHerbst** [comment]
@hoechenberger: this would overwrite itself when run multiple times? (I think it should)

**Comment by hoechenberger** [comment]
> @hoechenberger: this would overwrite itself when run multiple times? (I think it should)

Yes it would!

**Comment by hoechenberger** [comment]
But generally, any suggestions regarding motivation, file content, and phrasing? ;)

**Comment by hoechenberger** [comment]
This is good to go from my end :)

**Comment by agramfort** [comment]
thx @hoechenberger

**Pull request merged**